### PR TITLE
chore(doc) : Updates Tailwind Installation docs

### PR DESCRIPTION
**Pull request by danspratling**
## Description

- Adds support for 1.4.0
- Adds missing setup instructions for custom css/scss files
- Adds missing PurgeCSS documentation
- Minor navigation improvements

### Documentation

This PR is only updating documentation, it does not contain any new features which need to be further documented.

## Related Issues
#23952

### CODE REVIEW

File: docs/docs/tailwind-css.md [hunk 1]
```diff
@@ -149,6 +149,112 @@ plugins: [
...
+  ],
+  theme: {},
+  variants: {},
+  plugins: [],
+}
+```
+
+Full documentation on this can now be found on the Tailwind site - [Tailwind PurgeCSS documentation](https://tailwindcss.com/docs/controlling-file-size/#app)
+
+#### Older versions
```
Comment (KyleAMathews) [inline docs/docs/tailwind-css.md:218]: maybe we should only include docs for the newest version? I'd assume people wanting purgecss would either upgrade or find it directly.

Comment (danspratling) [reply docs/docs/tailwind-css.md:218]: I can remove. I added it as v1.4.0 only came out a week and a half ago, so most people will still be using the old approach. But you're right, it makes sense to direct people to the officially supported/recommended method.

Comment (danspratling) [reply docs/docs/tailwind-css.md]: @gatsbyjs/learning 
Updated as requested, removing all but the bare minimum support for older versions of TailwindCSS/PurgeCSS.

**Comment by danspratling** [comment]
@Ekwuno Done. The new issue can be found here https://github.com/gatsbyjs/gatsby/issues/23952 I've also linked it in the description above.

**Comment by Ekwuno** [comment]
> @Ekwuno Done. The new issue can be found here #23952 I've also linked it in the description above.

Thank you.

**Pull request merged**